packages/utils/src/utils/did.utils.ts

```diff
@@ -1,7 +1,7 @@
 import { assertNonNullish } from "./asserts.utils";
 
 export const toNullable = <T>(value?: T): [] | [T] => {
-  return value ? [value] : [];
+  return value !== undefined ? [value] : [];
```

Comment:
since we are already there, should we extend with `null`?

I'm not sure what happens if we send `[null]`. We never had the problem so it's not related to your PR but more asking myself because we are having a lookt at it.

Reply:
That's why I added the test for it. I consider it valid for now.

We probably want to talk with the team that devleops candid to make this decision.

Reply:
Just noticed I got following in my project.

```
export const toNullable = <T>(value?: T | null): [] | [T] => {
	return nonNullish(value) ? [value] : [];
};
```

We can do something similar.

Reply:
Do we know for sure that `[null]` is incorrect? What if the backend creates an optional field whose value is `null` or equivalent in motoko or rust?

Reply:
Almost sure but empirically. There is no `null` in Rust . `Option<Something>` converts to `[] | [Something]`.

Reply:
You're right! I just lookied into it. Even an option of an option would be most `[[]]`, not `[null]`. The yes, like you have it is perfect.

Reply:
https://github.com/dfinity/ic-js/pull/288